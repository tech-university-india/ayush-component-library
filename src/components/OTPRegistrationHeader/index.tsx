import React, { ChangeEvent } from "react"
import TextBox from "../TextBox"

interface OTPRegistrationHeaderProps {
  setFormData: (formDatay: any) => void
  disabled: boolean
  formData: any
}

function OTPRegistrationHeader ({ setFormData, disabled, formData }: OTPRegistrationHeaderProps) {
  return (
    <React.Fragment>
      <TextBox placeholder="Aadhaar Number" disabled={true} text={formData.aadhaarNumber} onChange={() => {
        // do nothing
      }}/>
      <p className="m-6 text-center">
        Please enter mobile number you want to use with ABHA
      </p>
      <TextBox placeholder="Mobile Number" text={formData.mobileNumber} onChange={(e:ChangeEvent<HTMLInputElement>) => setFormData({ ...formData, mobileNumber: e.target.value })} disabled={disabled}/>
    </React.Fragment>
  )
}

export default OTPRegistrationHeader
