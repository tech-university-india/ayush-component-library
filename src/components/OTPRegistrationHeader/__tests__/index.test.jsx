import React from "react"
import { render, screen } from "@testing-library/react"
import OTPRegistrationHeader from "../index.tsx"

const formData = {
  aadhaarNumber: "1234567890",
}

const setFormData = jest.fn()

describe("OTPRegistrationHeader", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<OTPRegistrationHeader formData={formData} setFormData={setFormData}/>)
    expect(asFragment()).toMatchSnapshot()
  })

  it("should render the correct text", () => {
    render(<OTPRegistrationHeader formData={formData} setFormData={setFormData}/>)
    expect(
      screen.getByText("Please enter mobile number you want to use with ABHA")
    ).toBeInTheDocument()
  })
})
