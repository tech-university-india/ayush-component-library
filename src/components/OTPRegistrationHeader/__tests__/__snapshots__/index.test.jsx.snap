// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OTPRegistrationHeader should render correctly 1`] = `
<DocumentFragment>
  <input
    class="disabled:text-preFillText p-3 w-full rounded-lg shadow-textBoxInset bg-textBox"
    disabled=""
    placeholder="Aadhaar Number"
    type="text"
    value="1234567890"
  />
  <p
    class="m-6 text-center"
  >
    Please enter mobile number you want to use with ABHA
  </p>
  <input
    class="disabled:text-preFillText p-3 w-full rounded-lg shadow-textBox bg-textBox"
    placeholder="Mobile Number"
    type="text"
    value=""
  />
</DocumentFragment>
`;
